iphone/cli/commands/_build.js

```diff
@@ -1936,7 +1936,11 @@ iOSBuilder.prototype.validate = function validate(logger, config, cli) {
 				const xcodeInfo = this.iosInfo.xcode;
 
 				function sortXcodeIds(a, b) {
-					return xcodeInfo[a].selected || appc.version.gt(xcodeInfo[a].version, xcodeInfo[b].version) ? -1 : appc.version.lt(xcodeInfo[a].version, xcodeInfo[b].version) ? 1 : 0;
+					// prioritize selected xcode
+					if (xcodeInfo[b].selected) {
+						return 1;
```

Comment:
You need to be mindful of the original sort order. This change makes the selected Xcode the last element where the previous code had it be the first. This sort function is used in 2 places. The first instance reverses the results and the other doesn't. This code would change that behavior.

`iosSdkVersion` is set if an iOS SDK version was passed in via the command line `--ios-version <ver>` option. When this is set, we want to pick the most recent Xcode or selected Xcode.

If there's no `iosSdkVersion` and it's a device/dist build, then we pick the oldest Xcode. Why? I have no idea. Compatibility? Who knows?

So, I'm thinking you either need to remove the `.reverse()` or change the `-1` to `1` and `1` to `-1` in the sort function. The former is probably the better, cleaner route. We should also figure out why the code chose the oldest Xcode to select an iOS Simulator from.